bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -665,6 +668,34 @@ public T setNettyMaxFrameSizeBytes(int maxSize) {
         return getThis();
     }
 
+    /**
+     * Timeout controlling wait on request send in case of unresponsive bookie(s)
+     * (i.e. bookie in long GC etc.)
+     *
+     * @return timeout value
+     *        negative value disables the feature
+     *        0 to allow request to fail immediately
+     *        Default is -1 (disabled)
+     */
+    public long getWaitTimeoutOnBackpressureMillis() {
```

Comment:
If it is from the client, how does the client know if the bookie is slow/dead/hung or something backpressure related? Also, how does this manifest with rest of the timeouts, say, read/write timeout

Reply:
all the client is see is that connection to bookie still exists but data cannot be send via it at the moment (or cannot be send fast enough to clean up existing backlog) and netty raises channel.isWritable() == false.
the rest is handled by client backpressure.